### PR TITLE
Make overflowed tag text ellipsis

### LINKED ISSUE #85: [BUG] - Saving a very long keyword extends past the extension UI boundaries
### Description

Saving a very long keyword extends past the extension UI boundaries. Should truncate and add a `...` ellipsis to the keyword when displaying the tag text.

### Reproduction steps

```bash
1. Open the extension
2. Click on 'Edit Allowed' button
3. Enter a very long keyword. E.g.'efafafafafafafaaffafafafaf'
4. Click on 'Save' button
5. See error
```

### Screenshots

截屏2022-09-25 下午8 45 24

### Logs

_No response_

### PR DESCRIPTION
**Pull request by Kanelmis**
## Todo

- [x] Fix tag item text overflow 

## Motivation

Saving a very long keyword extends past the extension UI boundaries. Should truncate and add a ... ellipsis to the keyword when displaying the tag text.

## Attached GitHub issue

Closes #85 

## Implementation
Only the CSS properties in `popup.css` were changed. Both `tag-item` and `.tag-results-item` needed changing as they both have the overflow UI bug. `tag-item` configures the CSS for most of the purple keyword chips. `.tag-results-item` configures the CSS for the chip that is used when there are too many keywords set e.g. "2 more". 

## Screenshot
截屏2022-09-28 下午12 57 21

### Documentation

- [x] Updated the readme documents (`README.md` etc.)
- [x] New functions and methods have additional comments added to document their usage

### Local Build

- [x] Ran all pre-commit hooks `pre-commit run --all-files`
- [x] Extension has been tested to build correctly `cd extension && yarn && yarn build`
- [x] Backend has been tested to build correctly `cd backend && docker-compose build`
- [x] Extension test suite has been run locally `cd extension && yarn && yarn test`
- [x] Backend test suite has been run locally `cd backend && python3 -m unittest`

### GitHub

- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

File: extension/public/popup.css [hunk 1]
```diff
@@ -241,12 +242,14 @@ header {
 
   border-radius: 12px;
   background-color: #c54ccf;
+
```
Comment (Destiny-02) [inline extension/public/popup.css:245]: Please remove extra empty lines as it is not related to this commit. If there is a convention that you would like to follow, we can discuss it in a meeting and create a code formatting issue.

Comment (Destiny-02) [inline extension/public/popup.css:244]: Please remove extra empty lines as it is not related to this commit. If there is a convention that you would like to follow, we can discuss it in a meeting and create a code formatting issue.

File: extension/public/popup.css [hunk 2]
```diff
@@ -233,6 +233,7 @@ header {
   padding: 0 12px 0 12px;
 
   height: 24px;
+  max-width: 148px;
```
Comment (Destiny-02) [inline extension/public/popup.css]: `max-width` is duplicated in 3 places in this PR. Are all instances needed?

File: extension/public/popup.css [hunk 3]
```diff
@@ -258,6 +261,15 @@ header {
   background-color: rgba(255, 255, 255, 0);
 }
 
+/* Tag Item's Text */
+.keywords .tag-list .tag-item .tagText{
```
Comment (Destiny-02) [inline extension/public/popup.css]: nit pick: `tagText` --> `tag-text` (coding convention this repository has adopted)

File: extension/public/popup.js [hunk 4]
```diff
@@ -101,10 +101,11 @@ const renderAllowedWords = () => {
         : "";
 
     for (let i = 0; i < numTagItems; i++) {
-      tagItems += `<div class="tag-item">${allowedWordsData[i]}</div>`;
+      tagItems += `<div class="tag-item" ><span class="tagText">${allowedWordsData[i]}</span></div>`;
     }
 
     tagList.innerHTML = tagItems + moreResults;
+
```
Comment (Destiny-02) [inline extension/public/popup.js]: Empty line. See comment above.

File: extension/public/popup.js [hunk 5]
```diff
@@ -101,10 +101,11 @@ const renderAllowedWords = () => {
         : "";
 
     for (let i = 0; i < numTagItems; i++) {
-      tagItems += `<div class="tag-item">${allowedWordsData[i]}</div>`;
+      tagItems += `<div class="tag-item" ><span class="tagText">${allowedWordsData[i]}</span></div>`;
```
Comment (Destiny-02) [inline extension/public/popup.js]: Can the `tagText` (or `tag-text`) go beside `tag-item` e.g. `class="tag-item tag-text`? This will reduce unnecessary complexity.

**Review by Destiny-02 — CHANGES_REQUESTED** [review]

File: extension/public/popup.js (hunk 5 above)
Comment (Kanelmis) [reply extension/public/popup.js]: If u mean can div class = "tag-item tag-text" then it would not work. Otherwise, do u expect me to edit the CSS file to match this change?

File: extension/public/popup.css (hunk 1 above)
Comment (Kanelmis) [reply extension/public/popup.css:244]: I am following the convention that they left in this CSS file. They separated the different parts of the CSS file by a empty line, though I have no idea why they do that,,,

**Comment by Kanelmis** [comment]
If we want to establish our code format convention, I'd like to make a change later.

File: extension/public/popup.css (hunk 1 above)
Comment (Destiny-02) [reply extension/public/popup.css:244]: This new line is just above a closing bracket and not close to the change you made on line 236.

File: extension/public/popup.js (hunk 5 above)
Comment (Destiny-02) [reply extension/public/popup.js]: What about `class="tag-item tagText`?

**Comment by Destiny-02** [comment]
> If we want to establish our code format convention, I'd like to make a change later.

Not 100% sure, but I think we are supposed to follow the conventions established by the previous group unless there is something obviously wrong that we have a good reason to drop. Otherwise, if you have spotted areas in the current codebase that don't follow their own conventions, a separate issue/PR can be made for this.

**Comment by Kanelmis** [comment]
> > If we want to establish our code format convention, I'd like to make a change later.
> 
> Not 100% sure, but I think we are supposed to follow the conventions established by the previous group unless there is something obviously wrong that we have a good reason to drop. Otherwise, if you have spotted areas in the current codebase that don't follow their own conventions, a separate issue/PR can be made for this.

Yes, that is why I left empty lines in CSS files

File: extension/public/popup.js [hunk 6]
```diff
@@ -101,10 +101,10 @@ const renderAllowedWords = () => {
         : "";
 
     for (let i = 0; i < numTagItems; i++) {
-      tagItems += `<div class="tag-item">${allowedWordsData[i]}</div>`;
+      tagItems += `<div class="tag-item tag-text">${allowedWordsData[i]}</div>`;
     }
 
-    tagList.innerHTML = tagItems + moreResults;
+    tagList.innerHTML = tagItems + moreResults;
```
Comment (Destiny-02) [inline extension/public/popup.js]: Can you revert this line? This PR should only affect only the lines that need to be modified. This will reduce the changes the reviewer needs to review and avoid unnecessary merge conflicts.

File: extension/public/popup.css [hunk 7]
```diff
@@ -258,6 +259,15 @@ header {
   background-color: rgba(255, 255, 255, 0);
 }
 
+/* Tag Item's Text */
+.keywords .tag-list .tag-text{
+    max-width: 124px;
+    text-overflow: ellipsis;
```
Comment (Destiny-02) [inline extension/public/popup.css]: nit pick: indentation. The indentation for these changes is different to the indentation in the rest of the files

**Comment by Kanelmis** [comment]
Early commits did not apply text-overflow to `tag-result-item`. It has been added in the end, because it should the same property as `tag-item` and `tag-result-item` both are items of tag lists

**Review by Destiny-02 — APPROVED** [review]

File: extension/public/popup.js (hunk 6 above)
Comment (Kanelmis) [reply extension/public/popup.js]: Done
